text/0000-hash/0000-hash.md

```diff
@@ -88,34 +88,46 @@ Useful links:
 
 # Detailed design
 
-This structure design is going to be the interface of any sponge function. Users can expect each hash instance would
-provide at least these public methods. Imagine it's like a real sponge, it `absorb`s the clone of input reference
-provided by users. And they don't need to know how internal permutation and transformation work, just `squeeze` the
-sponge and there will be a mutable reference as output. The internal state will not be cleared unless `reset` is called. 
+The sponges proposes in this RFC have a common interface, which consists of the functions below:
+
++ `fn absorb(&mut self, input: &[Trit])`;
++ `fn squeeze(&mut self) -> Vec<Trit>`;
++ `fn squeeze_into(&mut self, output: &mut [Trit])`;
++ `fn reset(&mut self)`;
++ `fn digest(&mut self, input: &[Trit]) -> Vec<Trit>`;
++ `fn digest_into(&mut self, input: &[Trit], output: &mut [Trit])`.
```

Comment:
I really like the additional methods. It adds more flexibility for users. But we will also have to consider how long outputs are. Either a fixed size same as hash length, or we should add a parameter for output length. I would prefer a fixed size length so the output could just be an array.

Reply:
I agree here, we need a size hint for methods like `fn squeeze(&mut self) -> Vec<Trit>`.